service/src/main/java/bio/terra/drshub/services/AuthService.java

```diff
@@ -0,0 +1,220 @@
+package bio.terra.drshub.services;
+
+import bio.terra.bond.model.SaKeyObject;
+import bio.terra.common.iam.BearerToken;
+import bio.terra.drshub.DrsHubException;
+import bio.terra.drshub.config.DrsProvider;
+import bio.terra.drshub.models.AccessUrlAuthEnum;
+import bio.terra.drshub.models.DrsHubAuthorization;
+import com.google.common.annotations.VisibleForTesting;
+import io.github.ga4gh.drs.model.AccessMethod;
+import io.github.ga4gh.drs.model.Authorizations;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.collections4.map.PassiveExpiringMap;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpStatusCodeException;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.util.UriComponents;
+
+@Service
+@Slf4j
+public class AuthService {
```

Comment:
`AuthService` is a new service to encapsulate anything that talks to an auth provider, like bond or ECM.